### DISCUSSION TITLE
HTML isolation failure causes code box to take effect

### DISCUSSION THREAD

**Discussion by NaturezzZ** [open-webui/open-webui#3732] [General]
# Bug Report

## Description

**Bug Summary:**
The code box does not isolate HTML code from being parsed by the browser, and it mistakenly takes effect.

**Steps to Reproduce:**

Step 1:

Type in the following prompt, and send it to LLM.
```
output the following contents in a code manner:
<!DOCTYPE html>
<html>
  <head>
    <title>Redirecting...</title>
    <meta http-equiv="refresh" content="5;url=http://localhost:3000">
  </head>
  <body>
    If you are not redirected automatically, follow this <a href="http://localhost:3000">link</a>.
  </body>
</html>
```

Step 2:

Type in `thank you` and send it to LLM.

Step 3:

Wait for a few seconds.

**Expected Behavior:**
The code is not executed.

**Actual Behavior:**
The browser is redirected to `localhost:3000` according to the code.

## Environment

- **Open WebUI Version:** v0.3.7
- **Ollama (if applicable):** 0.1.48

- **Operating System:** macOS 14.1
- **Browser (if applicable):** Chrome 126.0.6478.127

## Reproduction Details

You should type in `thank you` and wait for a few seconds after its response.
It consistently reproduces on Chrome, and appears less frequently on Safari.

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker

## Additional Information

None

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Investigating!

**Reply by tjbck** [reply]
Should be fixed on latest dev, let us know if the issue persists!

**Comment by NaturezzZ** [comment]
![open-webui-bug](https://github.com/open-webui/open-webui/assets/46887061/e4419ed3-e921-48f8-b3fa-85f29934661f)
buggy demo
